src/core_plugins/kibana/public/management/sections/indices/edit_index_pattern/indexed_fields_table/indexed_fields_table.js

```diff
@@ -1,92 +1,92 @@
+import { createSelector } from 'reselect';
 
-uiModules.get('apps/management')
-  .directive('indexedFieldsTable', function (Private, $filter) {
-    const yesTemplate = '<i class="fa fa-check" aria-label="yes"></i>';
-    const noTemplate = '';
-    const filter = $filter('filter');
-    const { fieldWildcardMatcher } = Private(FieldWildcardProvider);
+import { Table } from './components/table';
+import {
+  getFieldFormat
+} from './lib';
 
-    return {
-      restrict: 'E',
-      template,
-      scope: true,
-      link: function ($scope) {
-        const rowScopes = []; // track row scopes, so they can be destroyed as needed
-        $scope.perPage = 25;
-        $scope.columns = [
-          { title: 'name' },
-          { title: 'type' },
-          { title: 'format' },
-          { title: 'searchable', info: 'These fields can be used in the filter bar' },
-          { title: 'aggregatable', info: 'These fields can be used in visualization aggregations' },
-          { title: 'excluded', info: 'Fields that are excluded from _source when it is fetched' },
-          { title: 'controls', sortable: false }
-        ];
+export class IndexedFieldsTable extends Component {
+  static propTypes = {
+    indexPattern: PropTypes.object.isRequired,
+    fieldFilter: PropTypes.string,
+    indexedFieldTypeFilter: PropTypes.string,
+    helpers: PropTypes.shape({
+      redirectToRoute: PropTypes.func.isRequired,
+    }),
+    fieldWildcardMatcher: PropTypes.func.isRequired,
+  }
 
-        $scope.$watchMulti(['[]indexPattern.fields', 'fieldFilter', 'indexedFieldTypeFilter'], refreshRows);
+  constructor(props) {
+    super(props);
 
-        function refreshRows() {
-        // clear and destroy row scopes
-          _.invoke(rowScopes.splice(0), '$destroy');
-          const fields = filter($scope.indexPattern.getNonScriptedFields(), {
-            name: $scope.fieldFilter,
-            type: $scope.indexedFieldTypeFilter
-          });
-          const sourceFilters = $scope.indexPattern.sourceFilters && $scope.indexPattern.sourceFilters.map(f => f.value) || [];
-          const fieldWildcardMatch = fieldWildcardMatcher(sourceFilters);
-          _.find($scope.editSections, { index: 'indexedFields' }).count = fields.length; // Update the tab count
+    this.state = {
+      fields: []
+    };
+  }
+
+  componentWillMount() {
+    this.fetchFields();
+  }
+
+  mapFields(fields) {
+    const { indexPattern, fieldWildcardMatcher } = this.props;
+    const fieldWildcardMatch = fieldWildcardMatcher((indexPattern.sourceFilters && indexPattern.sourceFilters.map(f => f.value) || []));
 
-          $scope.rows = fields.map(function (field) {
-            const childScope = _.assign($scope.$new(), { field: field });
-            rowScopes.push(childScope);
+    return fields
+      .map((field) => {
+        return {
+          ...field,
+          displayName: field.displayName,
+          routes: field.routes,
+          indexPattern: field.indexPattern,
```

Comment:
lines 41-43 are covered by line 40.

Reply:
`displayName, routes, and indexPattern` aren't enumerable properties:
```js
Object.keys(field)
(7) ["name", "type", "count", "scripted", "searchable", "aggregatable", "readFromDocValues"]
```

Reply:
ahh, nm then :-)

Reply:
Can we leave a comment here explaining that? I'm pretty sure I would come along and make the same assumption Bill did. :)